libsolidity/ast/Types.cpp

```diff
@@ -167,8 +169,11 @@ void StorageOffsets::computeOffsets(TypePointers const& _types)
 	}
 	if (byteOffset > 0)
 		++slotOffset;
-	solAssert(slotOffset < bigint(1) << 256, "Object too large for storage.");
-	m_storageSize = u256(slotOffset);
+
+	solAssert(slotOffset < bigint(1) << 256, "Object extends past the end of storage.");
+	storageSize = slotOffset - _baseSlot;
+	solAssert(storageSize < bigint(1) << 256, "Object too large for storage.");
+	m_storageSize = u256(storageSize);
```

Comment:
You checked `slotOffset` already and `_baseSlot` is `u256`, so this assert is redundant.

```suggestion
	m_storageSize = u256(slotOffset - _baseSlot);
```

Reply:
You still have the `storageSize` variable in there. Just unused now.

Reply:
The unused variable is still there.

Reply:
Removed.